### ISSUE TITLE
try_parse fails to parse signed integer minimum values

### ISSUE THREAD

**Issue by danburkert** [Alexhuszagh/rust-lexical#10]
All of the following test cases fail with an `Overflow` error:

```rust
    assert_eq!(i8::MIN, lexical::try_parse(i8::MIN.to_string()).unwrap());
    assert_eq!(i16::MIN, lexical::try_parse(i16::MIN.to_string()).unwrap());
    assert_eq!(i32::MIN, lexical::try_parse(i32::MIN.to_string()).unwrap());
    assert_eq!(i64::MIN, lexical::try_parse(i64::MIN.to_string()).unwrap());
```

**Comment by Alexhuszagh** [comment]
Interesting, since I extensively test in lexical-core for this. Definitely a bug, just highlights how it's so easy to think you have better test coverage than you actually do. Thanks for finding this.

**Comment by Alexhuszagh** [comment]
Fixed in 3b1ebcd3bfac57b8ed2a4982391607e0a2a311e4. This was actually much more insidious than I initially imagined, thanks for catching this.

**Comment by Alexhuszagh** [comment]
Live as of [lexical v2.1](https://crates.io/crates/lexical/2.1.0) and [lexical-core v0.4](https://crates.io/crates/lexical-core/0.4.0).
